### ISSUE TITLE
Cube Designer is now even slower than before to display 10 lines of a big dataset

### ISSUE THREAD

**Issue by FabianCretton** [zazuko/cube-creator#1071]
My original issue about this is now closed: https://github.com/zazuko/cube-creator/issues/828

But today, I gave a try to the Cube Designer, and for the two cubes where it took up to 30 seconds, it now takes up to 1 minute:
https://int.cube-creator.lindas.admin.ch/app/cube-projects/cube-project!!fabforestrynormalized-5ljx6a5axdk/metadata
https://int.cube-creator.lindas.admin.ch/app/cube-projects/cube-project!!fab-test-forestry-3xrvo6r60jo/metadata

For the smaller one that took up to 10 seconds, it now takes up to 15:
https://int.cube-creator.lindas.admin.ch/app/cube-projects/cube-project!!fabianjahrbuch-wald-holz-2020test-2-xm3v67vbpkp/metadata

**Comment by l00mi** [comment]
Yes introducing the feature of paging and showing the amount of observation brought us back, and even added some time. 

We might investigate if can improve this in the future by separating these two issues.

**Comment by FabianCretton** [comment]
I was curious, so I tried a little query.
I see that requesting the first 10 observations is instantaneous without ORDER BY: https://s.zazuko.com/m7xBE
And then takes up to 38 seconds with an ORDER BY ?obs.
Is paging (OFFSET/LIMIT) possible without ORDER BY, using the natural order of triples in Stardog, or is it a random order ?

**Comment by l00mi** [comment]
The problem is not the ORDER BY, but to have a COUNT the DB needs to "touch" every entry, thus needs longer.

As said ideally we can separate this in two request steps. But not sure, because the ORDER BY is needed to assure that we don't get twice in a page the same values.

**Comment by FabianCretton** [comment]
The count to get the number of Observations takes 2-3 seconds, right -> https://s.zazuko.com/95Vnk

This is why I do not understand what was really added as new feature, is there anything else than the count of total triples ? 
I guess yes, as the time was not increased by 2-3 seconds, but 20-30.

Thank you to help me understand.

**Comment by FabianCretton** [comment]
By the way, thank you for adding the number of observations, I was waiting for that feature and it is really helpful
